aten/src/ATen/native/quantized/cpu/qclamp.cpp

```diff
@@ -41,6 +41,42 @@ Tensor quantized_clamp(
   return qy;
 }
 
+// hardtanh is clamp with default min==-1.0f and default max==1.0f
+Tensor quantized_hardtanh(
+    const Tensor& qx,
+    Scalar min,
+    Scalar max) {
+  Tensor qy;
+  AT_DISPATCH_QINT_TYPES(qx.scalar_type(), "hardtanh", [&]() {
+    qy = quantized_clamp_impl(qx, min, max);
```

Comment:
I think `quantized_clamp_impl` is in the anonymous namespace. Instead, you should use `at::clamp`

Reply:
[n00b] `qclamp` stub looks like its vectorized using `Vec256`, haven't benchmarked but I'm guessing it might be faster than `at::clamp`?

Reply:
[native_functions.yaml:L700](https://github.com/pytorch/pytorch/blob/eeda1146a5035bbc019c4dc01ea15966bdd4ab64/aten/src/ATen/native/native_functions.yaml#L700) says that `at::clamp` uses `quantized_clamp`, which in turn [uses `quantized_clamp_impl`](https://github.com/pytorch/pytorch/blob/eeda1146a5035bbc019c4dc01ea15966bdd4ab64/aten/src/ATen/native/quantized/cpu/qclamp.cpp#L39). So I think it should be the same, but without the use of anonymous namespace functions.

Reply:
Oh, my bad -- in this context you can use either `at::clamp` or `quantized_clamp_impl` -- it will do the same thing. I was confused. My bad!

Reply:
ah, ic, didn't realize at::clamp is before the dispatch, thanks!

for my own understanding, what's the downside of calling things which live in anonymous in pytorch?

Reply:
Sorry there is no downside -- I was too sleepy, and thought that you were trying to call an anonymous namespace from another file. This would cause an error. I just misread your code.